compiler/ccgtypes.nim

```diff
@@ -687,17 +697,25 @@ proc genRecordFieldsAux(m: BModule; n: PNode,
           result.addf("\t$1$3 $2;$n", [getTypeDescAux(m, field.loc.t, check, dkField), sname, noAlias])
   else: internalError(m.config, n.info, "genRecordFieldsAux()")
 
-proc genVirtualProcHeader(m: BModule; prc: PSym; result: var Rope; asPtr: bool = false, isFwdDecl:bool = false)
+proc genMemberProcHeader(m: BModule; prc: PSym; result: var Rope; asPtr: bool = false, isFwdDecl:bool = false)
 
 proc getRecordFields(m: BModule; typ: PType, check: var IntSet): Rope =
   result = newRopeAppender()
   genRecordFieldsAux(m, typ.n, typ, check, result)
-  if typ.itemId in m.g.graph.virtualProcsPerType:
-    let procs = m.g.graph.virtualProcsPerType[typ.itemId]
+  if typ.itemId in m.g.graph.memberProcsPerType:
+    let procs = m.g.graph.memberProcsPerType[typ.itemId]
+    var isDefaultCtorGen, isCtorGen: bool
     for prc in procs:
       var header: Rope
-      genVirtualProcHeader(m, prc, header, false, true)
+      if sfConstructor in prc.flags:
+        isCtorGen = true
+        if prc.typ.n.len == 1:
+          isDefaultCtorGen = true
+      genMemberProcHeader(m, prc, header, false, true)
       result.add "\t" & header & ";\n"
+    if isCtorGen and not isDefaultCtorGen:
+      var ch : IntSet
+      result.add "\t" & getTypeDescAux(m, typ, ch, dkOther) & "() = default;\n"
```

Comment:
I think you have to use `"$n"` instead of `"\n"` here because not all OS use the same line endings or something like that.

Reply:
No, there is a soft ($n) vs hard ($N) newline abstraction. Hard new lines ($N) are always kept, soft newlines can be skipped if activated for better debug info generation.

Reply:
Yes, thanks, what I mean is not to hardcode `"\n"`.

Reply:
@Araq do you see anything else I should change or you didn't go over it yet?
Some light on why testament is failing on win would be appreciated. That's all this PR is missing as far I can tell

Reply:
Ive seen random flaky tests on Windows, I wonder if we are occasionally hitting the max file open limit on Windows?.

Reply:
In local it outputs this before failing. It fails with a bunch of tests. 
```
warning: LF will be replaced by CRLF in C:\Users\juama\AppData\Local\Temp\diffStrings_b_FUEzC3aD.
The file will have its original line endings in your working directory
diff --git "a/C:\\Users\\juama\\AppData\\Local\\Temp\\diffStrings_a_YiVgiPft" "b/C:\\Users\\juama\\AppData\\Local\\Temp\\diffStrings_b_FUEzC3aD"
index 031aaaad3..88f2ffa0c 100644
--- "a/C:\\Users\\juama\\AppData\\Local\\Temp\\diffStrings_a_YiVgiPft"
+++ "b/C:\\Users\\juama\\AppData\\Local\\Temp\\diffStrings_b_FUEzC3aD"
@@ -1 +1,3 @@
-exitcode: 0
\ No newline at end of file
+exitcode: 1
+
+Output:
```